src/pypromice/process/utilities.py

```diff
@@ -148,22 +153,26 @@ def addMeta(ds, meta):
     ds : xarray.Dataset
         Dataset with metadata
    '''
-    ds['lon'] = ds['gps_lon'].mean()
-    ds['lon'].attrs = ds['gps_lon'].attrs
+    if 'gps_lon' in ds.keys():
+        ds['lon'] = ds['gps_lon'].mean()
+        ds['lon'].attrs = ds['gps_lon'].attrs
 
-    ds['lat'] = ds['gps_lat'].mean()
-    ds['lat'].attrs = ds['gps_lat'].attrs
+        ds['lat'] = ds['gps_lat'].mean()
+        ds['lat'].attrs = ds['gps_lat'].attrs
 
-    ds['alt'] = ds['gps_alt'].mean()
-    ds['alt'].attrs = ds['gps_alt'].attrs
+        ds['alt'] = ds['gps_alt'].mean()
+        ds['alt'].attrs = ds['gps_alt'].attrs
 
     # for k in ds.keys(): # for each var
     #     if 'units' in ds[k].attrs:
     #         if ds[k].attrs['units'] == 'C':
     #             ds[k].attrs['units'] = 'degrees_C'
 
     # https://wiki.esipfed.org/Attribute_Convention_for_Data_Discovery_1-3#geospatial_bounds
-    ds.attrs['id'] = 'dk.geus.promice:' + str(uuid.uuid3(uuid.NAMESPACE_DNS, ds.attrs['station_id']))
+    if 'station_id' in ds.attrs.keys():
+        ds.attrs['id'] = 'dk.geus.promice:' + str(uuid.uuid3(uuid.NAMESPACE_DNS, ds.attrs['station_id']))
+    else:
+        ds.attrs['id'] = 'dk.geus.promice:' + str(uuid.uuid3(uuid.NAMESPACE_DNS, ds.attrs['site_id']))
```

Comment:
>Is it correct that populateMeta is invoked at level 0?

`populateMeta` is indeed called only on the creation of the L0 objects.

>Do all L0 datasets include both station_id and site_id?

No L0 are station-specific objects, so they do not have a site_id at the moment.

I admit that I do not understand what is this `uuid` and whether it gives a unique string for each call or for each input station/site id (in which case it the uuid is the same for the station 'DY2' and the site 'DY2')? 
I also don't understand how it is being used by the users.

Reply:
**Now**:
- I think you should keep the station ID-based ID generation at L0 as it is for now.
- The site-based ID should instead be created when exporting the datasets.

**Future**:
- Postponing the ID generation also makes it possible to create individual IDs for various sample methods. I have created an issue describing an update to multiple metadata attributes: [GitHub Issue #259](https://github.com/GEUS-Glaciology-and-Climate/pypromice/issues/259).

Reply:
I am not sure I understand your suggestion.
Right now the ID is only derived before a dataset is written to file.
You mentioned `populateMeta` earlier, but  `populateMeta` only adds to a L0 dataset the attributes contained in the related config file, not anything that is produced in `addMeta`.

I also don't understand why we add a uuid? Is that number different every time a new file is generated (meaning every hour)?
or is it a number that is linked to a station_id? In which case I don't see why we would replace a human readable stid by a lengthy uuid.

Reply:
I have suggest using the following:

https://github.com/GEUS-Glaciology-and-Climate/pypromice/blob/7756f3062ba3606ec7ae965ff8f0932165616abb/src/pypromice/process/write.py#L258-L278

Reply:
`uuid3` is a hash function that generates a 128-bit number from an input string, designed to be globally unique. The output depends solely on the input string (and namespace) and will always return the same value for the same input.

Reply:
I think your suggestion for the ID is much better compared to the previous one, as it distinguishes between sample rates.

We should also consider:
1. IDs for source level datasets like tx and raw.
2. IDs for different levels. Station datasets can be stored at both level 2 and level 3. Maybe level 3 could be implicit since it is the official output level.
3. Making IDs unique for each iteration of a dataset. This makes it possible to precisely refer to the actual data used for analysis and processing. We should use dataset IDs extensively in our pipeline to determine whether an output has already been processed. We can use information about the input datasets, `pypromice` version, etc., to make the iteration ID deterministic.

`uuid3` is a hash function that generates a 128-bit number from an input string, designed to be globally unique. The output depends solely on the input string (and namespace) and will always return the same value for the same input. A benefit of using a hash function for the IDs is to control and limit the format of the ID string. This might be especially relevant for point (3).

Reply:
Thanks!
I'll stick to this version for this PR but I might add the level and the format in the next PRs.
Got it for the uid3. It might be a good way of boiling down many informations into a single id, but it don't think we're getting submerged by metadata at the moment.